File: cpp/009_PalindromeNumber.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if( x < 0 ) return false;
        int h = 1;
        int l = 1;
        int t = x;
        while( t >= 10){
            t /= 10;
            h *= 10;
        }
        
        while(h > l){
            if((x/h)%10 != (x/l)%10) return false;
            h /= 10;
            l *= 10;
        }
        return true;
    }
};

File: js/009_PalindromeNumber.js
/**
 * @param {number} x
 * @return {boolean}
 */
var isPalindrome = function(x) {
    if( x < 0 ) return false;
    var h = 1;
    var l = 1;
    var t = x;
    while( t >= 10 ){
    	t = Math.floor( t/ 10);
    	h *= 10;
    }

    while( h > l ){
    	if( Math.floor(x/h)%10 != Math.floor(x/l)%10){
    		return false;
    	}
    	h = Math.floor( h/10 );
    	l *= 10;
    }
    return true;
};